javalib/src/main/scala/java/net/HttpCookie.scala

```diff
@@ -0,0 +1,413 @@
+          char == ' ' ||
+          char == '\t') {
+        throw new IllegalArgumentException("Illegal cookie name")
+      }
+
+      i += 1
+    }
+
+    name
+
+  }
+
+}
+
+/** HttpCookie */
+final class HttpCookie private (
+    private val createdAtEpochSecond: Long,
+    private var _comment: String = null,
+    private var _commentURL: String = null,
+    private var _httpOnly: Boolean = false,
+    private var _discard: Boolean = false,
+    private var _domain: String = null,
+    private var _maxAge: Long = -1,
+    private var _name: String = null,
+    private var _path: String = null,
+    private var _portlist: String = null,
+    private var _secure: Boolean = false,
+    private var _value: String = null,
+    private var _version: Int = 1
+) extends Cloneable {
+
+  def this(name: String, value: String) = {
+    this(
+      System.currentTimeMillis() % 1000,
+      _name =
+        (if (name.length() == 0) {
+           throw new IllegalArgumentException("Illegal cookie name")
+         } else
+           HttpCookie.validateName(name)),
+      _value = value
+    )
+  }
+
+  def canEqual(other: Any): Boolean =
+    other.isInstanceOf[HttpCookie]
+
+  override def hashCode(): Int = {
+    var res = 1
+    // the JVM seems to include
+    // the name, value, domain and path into its
+    // hash code, so that's what we do as well
+    res = 31 * res + _name.##
+    if (_value != null) res = 31 * res + _value.##
+    if (_path != null) res = 31 * res + _path.##
+    if (_domain != null) res = 31 * res + _domain.##
+    res
+  }
+
+  override def toString(): String = _name + "=" + _value
+  override def equals(obj: Any): Boolean =
+    obj match {
+      case other: HttpCookie =>
+        super.equals(other) &&
+          canEqual(other) &&
+          this.getName() == other.getName() &&
+          this.getValue() == other.getValue() &&
+          this.getComment() == other.getComment() &&
+          this.getCommentURL() == other.getCommentURL() &&
+          this.isHttpOnly() == other.isHttpOnly() &&
+          this.getDiscard() == other.getDiscard() &&
+          this.getDomain() == other.getDomain() &&
+          this.getMaxAge() == other.getMaxAge() &&
+          this.getPath() == other.getPath() &&
+          this.getPortlist() == other.getPortlist() &&
+          this.getSecure() == other.getSecure() &&
+          this.getVersion() == other.getVersion()
+
+      case _ => false
+    }
+
+  def getComment(): String = this._comment
+  def getCommentURL(): String = this._commentURL
+  def getDiscard(): Boolean = this._discard
+  def getDomain(): String = this._domain
+  def getMaxAge(): Long = this._maxAge
+  def getName(): String = this._name
+  def getPath(): String = this._path
+  def getPortlist(): String = this._portlist
+  def getSecure(): Boolean = this._secure
+  def getValue(): String = this._value
+  def getVersion(): Int = this._version
+  def hasExpired(): Boolean = {
+    _maxAge >= 0 && ((System
```

Comment:
JVM would return true if `maxAge == 0`, so we should start with 2 quick range checks before starting getting current time